test/e2e/common/storage/projected_clustertrustbundle.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+
+	certificatesv1alpha1 "k8s.io/api/certificates/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epodoutput "k8s.io/kubernetes/test/e2e/framework/pod/output"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"k8s.io/utils/ptr"
+
+	"github.com/onsi/ginkgo/v2"
+)
+
+var _ = SIGDescribe("Projected clustertrustbundle [Feature:ClusterTrustBundle] [Feature:ClusterTrustBundleProjection]", func() {
+	f := framework.NewDefaultFramework("projected-clustertrustbundle")
+	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
+
+	goodCert1 := mustMakeCertificate(&x509.Certificate{
+		SerialNumber: big.NewInt(0),
+		Subject: pkix.Name{
+			CommonName: "root1",
+		},
+		IsCA:                  true,
+		BasicConstraintsValid: true,
+	})
+
+	goodCert1Block := string(mustMakePEMBlock("CERTIFICATE", nil, goodCert1))
+
+	ginkgo.It("should be able to mount a single ClusterTrustBundle by name", func(ctx context.Context) {
```

Comment:
Add more test cases like via signer name, signer name with and without label selector, empty but not optional, empty and optional, etc.  I recommend doing that with ~100s of CTBs to emulate the worse case.

Reply:
I can't figure out how to run this, so I didn't want to add a bunch of test cases.  My current plan is to land it, so that it goes into the gce/alpha CI test set, and then I can iterate on adding cases.

Reply:
@ahmedtd did you confirm that CI is at least running the new e2e?

Reply:
It is not yet; it has to exist in k/k before it can be turned on in k/test-infra, is my understanding.